advanced_survey/static/advanced-survey/src/direflow-components/field-selector/index.tsx

```diff
@@ -79,12 +81,16 @@ const FieldSelector: FC<IProps> = ({index, question}) => {
                     </div>
                     <div className="form-group mb-2">
                         <label>Choices:</label>
-                        <textarea className="form-control" value={options.choices} onChange={e => setOption("choices", e.target.value)}></textarea>
+                        <OptionsWidget options={options.choices} changeHandler={(ops: string[])=> setOption("choices", ops)} />
```

Comment:
Could we add a `useCallback` hook for the change handler to prevent re-renders?

Reply:
What is dependency of `useCallback ` ?

Reply:
I would say

```
    const setOption = useCallback((property: string, value: any) => {
        setOptions((old: any) => {
            return {...old, [property]: value};
        })
    }, [])
```

And then create separate callbacks for each property (multiple, max, url...) but I guess that would be a lot of separate callbacks. Could also have `setOption` accept an event as an argument and have behaviour change based on `event.target.id`.

But writing this out it sounds unnecessary. If you don't think it's worth doing, go ahead and merge